File: scripts/refactor_tools.go
package main

import (
	"io/fs"
	"os"
	"path/filepath"
	"strings"
)

func main() {
	filepath.WalkDir("internal/tools", func(path string, d fs.DirEntry, err error) error {
		if !d.IsDir() && strings.HasSuffix(path, ".go") {
			content, _ := os.ReadFile(path)
			text := string(content)

			// Remove imports
			text = strings.ReplaceAll(text, "\"github.com/voocel/agentcore/schema\"", "")
			text = strings.ReplaceAll(text, "\"github.com/voocel/agentcore/tools\"", "")

			// Comment out Schema() methods (hacky but works for now, we'll fix compile errors later)
			lines := strings.Split(text, "\n")
			inSchema := false
			for i, line := range lines {
				if strings.Contains(line, ") Schema() map[string]any") {
					inSchema = true
				}
				if inSchema {
					lines[i] = "// " + line
					if strings.HasPrefix(line, "}") {
						inSchema = false
					}
				}
			}

			os.WriteFile(path, []byte(strings.Join(lines, "\n")), 0644)
		}
		return nil
	})
}

File: scripts/refactor_tools.py
import os

def main():
    base_dir = 'internal/tools'
    for root, dirs, files in os.walk(base_dir):
        for file in files:
            if file.endswith('.go'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    text = f.read()
                
                text = text.replace('"github.com/voocel/agentcore/schema"', '')
                text = text.replace('"github.com/voocel/agentcore/tools"', '')
                
                lines = text.split('\n')
                in_schema = False
                for i, line in enumerate(lines):
                    if ') Schema() map[string]any' in line:
                        in_schema = True
                    if in_schema:
                        lines[i] = '// ' + line
                        if line.startswith('}'):
                            in_schema = False
                
                with open(path, 'w', encoding='utf-8') as f:
                    f.write('\n'.join(lines))

if __name__ == '__main__':
    main()
